homeassistant/components/threshold/binary_sensor.py

```diff
@@ -199,15 +209,21 @@ def extra_state_attributes(self):
         }
 
     @callback
-    def _update_state(self):
+    def _update_state(self) -> None:
         """Update the state."""
 
-        def below(threshold):
+        def below(threshold: float | None) -> bool | None:
             """Determine if the sensor value is below a threshold."""
+            if TYPE_CHECKING:
+                assert threshold is not None
+                assert self.sensor_value is not None
```

Comment:
IMHO this is now needed because of weird/incorrect handling that should be fixed instead.

Let's fix the signature of this method and guard it against being called with a 0 value.

`self.sensor_value` seems to be possible to be `None` and isn't guarded.